### PR TITLE
fix(balances): query correct balance endpoint, show microblock balance

### LINKED ISSUE #2898: Resolve delay in recognizing confirmed transactions
Here transaction in microblock
image

Yet on wallet shown as available

trying to sign a transaction will fail because the balance isn't actually available:
image

This is happening on mainnet too:
https://discord.com/channels/621759717756370964/745197302255321108/1044462352642363412

Additionally when the transaction was confirmed in an anchorblock. I tried to send the transaction again from the btc.us testenvironment. And it would still show "insufficient" balance error on the confirm screen. I was only able to resolve this after logging out from the dapp and then logging in again, then the confirm screen still blinked "insufficient balance" for a split second before showing the expected:
image

### PR DESCRIPTION
**Pull request by kyranjamie**
> Try out this version of the Hiro Wallet - download [extension builds](https://github.com/hirosystems/stacks-wallet-web/actions/runs/3525782388).

- Fixes endpoint named anchored but querying unanchored balance
- Fixes microblock balance not showing on homepage
- Displays anchored balance in transaction request header
- Displays anchored balance in choose/switch account preview
- Increasing query time for balance endpoint, was far too slow

cc/ @314159265359879

### CODE REVIEW

**Comment by 314159265359879** [comment]
On this build I see it fixed at the top:
image

But I still see 500 STX balance while in microblock on account choose/switch screen:
image

And the dapp does let me send the transaction as if I have 500 STX anchored.

File: src/app/query/stacks/balance/crypto-asset-balances.hooks.ts [hunk 1]
```diff
@@ -26,6 +26,14 @@ export function useStacksAnchoredCryptoCurrencyAssetBalance(address: string) {
       ),
   });
 }
+export function useStacksUnanchoredCryptoCurrencyAssetBalance(address: string) {
+  return useUnanchoredStacksAccountBalanceQuery(address, {
+    select: resp =>
```
Comment (edu-stx) [inline src/app/query/stacks/balance/crypto-asset-balances.hooks.ts:31]: :+1:

**Review by edu-stx — APPROVED** [review]
Looks good!

**Comment by fbwoolf** [comment]
~~I don't understand your list item here: `Fixes endpoint named anchored but querying unanchored balance`? It was never named incorrectly as `anchored`? There is an existing `useStacksClientAnchored`. The one you are referring to was correct but you just added the name `unanchored` to it? Am I missing something here?~~

Nvm: I ended up seeing what you meant. I read the bullet point incorrectly.

File: src/app/features/balances-list/balances-list.tsx [hunk 2]
```diff
@@ -54,8 +57,13 @@ export const BalancesList = ({ address, ...props }: BalancesListProps) => {
       {btcAssetBalance.balance.amount.isGreaterThan(0) && (
         <CryptoCurrencyAssetItem assetBalance={btcAssetBalance} icon={<Box as={BtcIcon} />} />
       )}
-      {stxAssetBalance.balance.amount.isGreaterThan(0) && (
-        <CryptoCurrencyAssetItem assetBalance={stxAssetBalance} icon={<StxAvatar {...props} />} />
+      {(stxAssetBalance.balance.amount.isGreaterThan(0) ||
+        stxUnachoredAssetBalance.balance.amount.isGreaterThan(0)) && (
```
Comment (fbwoolf) [inline src/app/features/balances-list/balances-list.tsx:61]: Why is this needed here now, bc you removed subBalance?

File: src/app/pages/choose-account/components/accounts.tsx [hunk 3]
```diff
@@ -64,7 +63,7 @@ const ChooseAccountItem = memo((props: ChooseAccountItemProps) => {
   const [component, bind] = usePressable(true);
   const { decodedAuthRequest } = useOnboardingState();
   const name = useAccountDisplayName(account);
-  const { data: balances, isLoading: isBalanceLoading } = useUnanchoredStacksBalances(
+  const { data: balances, isLoading: isBalanceLoading } = useAnchoredStacksAccountBalances(
```
Comment (fbwoolf) [inline src/app/pages/choose-account/components/accounts.tsx:67]: Need to import here.

File: src/app/features/balances-list/balances-list.tsx (hunk 2 above)
Comment (kyranjamie) [reply src/app/features/balances-list/balances-list.tsx:61]: Unrelated, as previously still only the one value was checked.

File: src/app/query/stacks/balance/balance.query.ts [hunk 4]
```diff
@@ -47,18 +48,19 @@ export function useAnchoredStacksAccountBalanceQuery<T extends unknown = FetchAc
...
-    retryOnMount: true,
+
     ...balanceQueryOptions,
     ...options,
   });
 }
 
 export function useAnchoredAccountBalanceListQuery(accounts?: AccountWithAddress[]) {
-  const client = useStacksClient();
+  const client = useStacksClientAnchored();
```
Comment (fbwoolf) [inline src/app/query/stacks/balance/balance.query.ts:63]: This is weird if causing bug now bc I looked back at my PR refactoring to the `StacksClient` and `BitcoinClient`, and this was using `useApi` (unanchored) before too.

EDIT: Not important to respond to, just refactoring must have exposed it in some way.

Comment (kyranjamie) [reply src/app/query/stacks/balance/balance.query.ts:63]: 🤷🏼  I don't think this was causing the bug, but the hook is named achored and was using unachored

**Review by fbwoolf — APPROVED** [review]

**Comment by kyranjamie** [comment]
Updated and fixed Werner's comment

**Review by edu-stx — APPROVED** [review]
